src/Raven.Studio/typescript/common/queryCompleter.ts

```diff
@@ -468,24 +461,35 @@ class queryCompleter {
             default:
                 break;
         }
+        
+        return lastKeyword;
     }
 
-    private completeWords(callback: (errors: any[], wordList: autoCompleteWordList[]) => void, keywords: ({value: string; score: number; meta: string})[]) {
+    private completeWords(callback: (errors: any[], wordList: autoCompleteWordList[]) => void, keywords: autoCompleteWordList[]) {
         callback(null,  keywords.map(keyword  => {
-            const word = <autoCompleteWordList>keyword;
-            word.caption = _.trim(keyword.value, "'");
+            if (!keyword.caption){
+                keyword.caption = _.trim(keyword.value, "'");
+            }
             if (keyword.meta === "function"){
                 keyword.value += "(";
             } else {
                 keyword.value += " "; // insert space after each completed keyword or other value.
             }
-            return word;
+            return keyword;
         }))
     }
 
-    private isLetterOrDigit(str: string) {
-        // TODO: Add support for more letters in other lanuagse.
-        return /^[0-9a-zA-Z_@]+$/.test(str)
+    private escapeCollectionOrFieldName(name: string) : string {
```

Comment:
this can be static method, right?

Reply:
I changed it to be static method.
But in general I do not like the static syntax of typescript so this is why I used instance method here by default.

Reply:
I agree that syntax for static method in TS is weird. :(